probe/process/reporter.go

```diff
@@ -0,0 +1,58 @@
+package process
+
+import (
+	"strconv"
+
+	"github.com/weaveworks/scope/report"
+)
+
+// We use these keys in node metadata
+const (
+	PID     = "pid"
+	Comm    = "comm"
+	PPID    = "ppid"
+	Cmdline = "cmdline"
+)
+
+// Reporter generate Reports containing the Process topology
+type Reporter struct {
+	procRoot string
+	scope    string
+}
+
+// NewReporter makes a new Reporter
+func NewReporter(procRoot, scope string) *Reporter {
+	return &Reporter{
+		procRoot: procRoot,
+		scope:    scope,
+	}
+}
+
+// Report generates a Report containing the Process topology
+func (r *Reporter) Report() (report.Report, error) {
+	result := report.MakeReport()
+	processes, err := r.processTopology()
+	if err != nil {
+		return result, err
+	}
+	result.Process.Merge(processes)
+	return result, nil
+}
+
+func (r *Reporter) processTopology() (report.Topology, error) {
+	t := report.NewTopology()
+	err := Walk(r.procRoot, func(p *Process) {
+		pidstr := strconv.Itoa(p.PID)
+		nodeID := report.MakeProcessNodeID(r.scope, pidstr)
+		t.NodeMetadatas[nodeID] = report.NodeMetadata{
```

Comment:
Oh! it's a map. Looks like a struct. Adds a bit of type-safety, but a bit non-obvious.